apps/sms/views/conversation/js/conversation.js

```diff
@@ -2973,52 +2837,51 @@ var ConversationView = {
     return defer.promise;
   },
 
-  discardDraft: function conv_discardDraft() {
-    // If we were tracking a draft properly, update the Drafts object entry.
-    if (!this.draft) {
-      return;
-    }
+  updateDraft() {
+    var conversation = this.activeThread;
 
-    Drafts.delete(this.draft).store();
+    // Need to assimilate recipients in order to check if any entered.
+    this.assimilateRecipients();
 
-    this.draft = null;
-  },
+    // If message content is not empty and/or we have _valid_ recipients in
+    // NewMessage view.
+    var hasDraftToSave = !Compose.isEmpty() ||
+      (!conversation && this.recipients.numbers.length);
 
-  /**
-   * Saves currently unsent message content and/or recipients into a Draft
-   * object. Preserves currently active draft if specified. Draft preservation
-   * is intended to keep this.draft populated with the currently active draft
-   * when app is hidden, so that when app becomes visible again, it knows there
-   * is a draft to continue to keep track of.
-   *
-   * @param {boolean?} preserveDraft Optional boolean parameter that indicates
-   * whether or not we should preserve draft.
-   */
-  saveDraft: function conv_saveDraft(preserveDraft) {
-    var thread = this.activeThread;
+    // Check if conversation has been just removed, can happen when user selects
+    // all messages in Conversation view and removes all of them, in this case
+    // we don't want to loose draft.
```

Comment:
ah no, I get it, we save it back later on. Maybe comment this: `we'll delete the conversation-bound draft and create a "new message" draft instead`